flatnav/util/SquaredL2SimdExtensions.h

```diff
@@ -45,11 +77,23 @@ static float computeL2_Avx2(const void *x, const void *y,
     v2.loadu(pointer_y);
     difference = v1 - v2;
     sum += difference * difference;
+
+    pointer_x += 8;
+    pointer_y += 8;
+
+    v1.loadu(pointer_x);
+    v2.loadu(pointer_y);
+    difference = v1 - v2;
+    sum += difference * difference;
+
     pointer_x += 8;
     pointer_y += 8;
   }
 
-  return sum.reduce_add();
+  float result[8];
+  sum.storeu(result);
+  return result[0] + result[1] + result[2] + result[3] + result[4] + result[5] +
+         result[6] + result[7];
```

Comment:
I noticed that this arduous computation is actually a few milliseconds faster than using `sum.reduce_add()` for AVX (although not true for SSE). So, that's why I'm using it here instead of the custom-made `reduce_add`.